### PR TITLE
Add openweathermap example app

### PR DESCRIPTION
**Pull request by briandowns**
Signed-off-by: Brian Downs <brian.downs@gmail.com>

### Well detailed description of the change :

Adds example code using the HTTP, System, Env, and JSON modules to connect to, parse, and print a response from openweathermap.org.

Resolves: #

#

### Type of change:

- [ ] Bug fix

- [ ] New feature

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#

### Housekeeping

- [ ] Tests have been updated to reflect the changes done within this PR (if applicable).

- [ ] Documentation has been updated to reflect the changes done within this PR (if applicable).

#

### CODE REVIEW

File: examples/openweathermap.du [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    current(name) {
+        var url = BASE_URL;
+        url = url + "appid=" + this.token;
+        url = url + "&q=" + name;
+        url = url + "&units=" + this.units;
+        url = url + "&lang=" + this.lang;
+
+        var res = HTTP.get(url);
```
Comment (Jason2605) [inline examples/openweathermap.du:29]:
```suggestion
        const res = HTTP.get(url);
```

File: examples/openweathermap.du [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        return JSON.parse(res.unwrap()["content"]).unwrap();
+    }
+}
+
+// main
+{
+    const token = Env.get("OWM_TOKEN");
+    if (token == "" or not token) {
```
Comment (Jason2605) [inline examples/openweathermap.du:37]: An empty string is considered "falsey" so all we need is the `not` call!

```suggestion
    if (not token) {
```

**Review by Jason2605 — CHANGES_REQUESTED** [review]
More incredibly nitpick comments sorry!

Also if we add examples to the `runExamples.du` file, it'll make sure they don't become outdated: https://github.com/dictu-lang/Dictu/blob/develop/examples/runExamples.du

**Comment by briandowns** [comment]
This example is dependent on a valid token being available set as an env var. Not sure if we'd want to add it to `runExamples.du`.

**Comment by Jason2605** [comment]
> This example is dependent on a valid token being available set as an env var. Not sure if we'd want to add it to `runExamples.du`.

Fair point yep!

**Pull request merged**